src/mixins/MemoryMixin.h

```diff
@@ -126,10 +126,11 @@ class WithIntrusiveDeleter {
 
 } // namespace details
 
-template <typename Base, typename... TArgs>
+template <typename Base, typename Payload = Payload, typename... TArgs>
 Base& createManagedInstance(TArgs&&... args) {
   auto* instance =
-      new MemoryMixin<typename details::WithIntrusiveDeleter<Base>::T>(
+      new MemoryMixin<typename details::WithIntrusiveDeleter<Base>::T,
+                      Payload>(
```

Comment:
Actually, when you run this with

> cmake -DCMAKE_BUILD_TYPE=DEBUG

I get
```
  ==7521==ERROR: LeakSanitizer: detected memory leaks
```
This is true even before the changes this diff introduces.

More generally, I think something like
```
return *(new MyClass);
```
leads to an awkward API and memory model that's difficult to manage.